catalog/core/catalog-core-contentresourcereader/src/main/java/org/codice/ddf/catalog/content/resource/reader/DerivedContentActionProvider.java

```diff
@@ -0,0 +1,115 @@
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.data.Metacard;
+
+public class DerivedContentActionProvider implements ActionProvider {
+
+    private static final String ID = "catalog.data.metacard.derived-content";
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(DerivedContentActionProvider.class);
+
+    private final ActionProvider resourceActionProvider;
+
+    private static final String DESCRIPTION_PREFIX = "Retrieves derived resource: ";
+
+    public DerivedContentActionProvider(ActionProvider actionProvider) {
+        this.resourceActionProvider = actionProvider;
+    }
+
+    @Override
+    public <T> List<Action> getActions(T input) {
+        if (!canHandle(input)) {
+            return Collections.emptyList();
+        }
+        // Expect only 1
+        List<Action> resourceActions = resourceActionProvider.getActions(input);
+        if (resourceActions.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<Action> actions = new ArrayList<>();
+        ((Metacard) input).getAttribute(Metacard.DERIVED_RESOURCE_URI)
+                .getValues()
+                .stream()
+                .forEach(value -> {
+                    try {
+                        URI uri = new URI(value.toString());
+                        URIBuilder builder = new URIBuilder(resourceActions.get(0)
+                                .getUrl()
+                                .toURI());
+                        if (StringUtils.equals(uri.getScheme(), ContentItem.CONTENT_SCHEME)) {
+                            String qualifier = uri.getFragment();
+
+                            builder.addParameters(Arrays.asList(new BasicNameValuePair(ContentItem.QUALIFIER,
+                                    qualifier)));
+                            ActionImpl newAction = new ActionImpl(ID,
+                                    "View " + qualifier,
+                                    DESCRIPTION_PREFIX + qualifier,
+                                    builder.build()
+                                            .toURL());
+                            actions.add(newAction);
+                        } else {
+                            ActionImpl newAction = new ActionImpl(ID,
+                                    "View " + uri.toString(),
+                                    DESCRIPTION_PREFIX + uri.toString(),
+                                    uri.toURL());
+                            actions.add(newAction);
+                        }
+                    } catch (URISyntaxException | MalformedURLException e) {
+                        LOGGER.debug("Unable to create action URL.", e);
+                    }
+                });
+        return actions;
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public <T> boolean canHandle(T subject) {
+        if (subject != null && Metacard.class.isAssignableFrom(subject.getClass())) {
```

Comment:
Couldn't this be simplified to `if (subject instanceof Metacard)`? This will handle null as well.